assets/js/components/dashboard-sharing/DashboardSharingSettingsButton.js

```diff
@@ -0,0 +1,131 @@
+/**
+ * DashboardSharingSettingsButton component.
+ *
+ * Site Kit by Google, Copyright 2022 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies
+ */
+import Dialog, { DialogContent, DialogFooter } from '@material/react-dialog';
+
+/**
+ * WordPress dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import { Icon, arrowLeft } from '@wordpress/icons';
+import {
+	createInterpolateElement,
+	Fragment,
+	useCallback,
+	useState,
+} from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import Link from '../Link';
+import Button from '../Button';
+import Portal from '../Portal';
+import ShareIcon from '../../../svg/icons/share.svg';
+import { BREAKPOINT_SMALL, useBreakpoint } from '../../hooks/useBreakpoint';
+
+export default function DashboardSharingSettingsButton() {
+	const breakpoint = useBreakpoint();
+	const [ dialogOpen, setDialogOpen ] = useState( false );
+
+	const openDialog = useCallback( () => {
+		setDialogOpen( true );
+	}, [] );
+
+	const closeDialog = useCallback( () => {
+		setDialogOpen( false );
+	}, [] );
+
+	return (
+		<Fragment>
+			<Link
+				aria-label={ __( 'Open sharing settings', 'google-site-kit' ) }
+				onClick={ openDialog }
+			>
+				<ShareIcon width={ 21 } height={ 21 } />
+			</Link>
+
+			<Portal>
+				<Dialog open={ dialogOpen } className="googlesitekit-dialog">
```

Comment:
@asvinb, we also need to add `onClose={ closeDialog }` property.

Reply:
ah! good catch! Updated!